src/App.tsx

```diff
@@ -1,36 +1,55 @@
-import React from 'react';
+import React, { useState, useMemo } from 'react';
 import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import moviesFromServer from './api/movies.json';
 
 export const App: React.FC = () => {
+  const [query, setQuery] = useState('');
+  const [visibleMovies, setVisibleMovies] = useState(moviesFromServer);
+
+  const filterMovies = (searchQuery: string) => {
+    const filteredMovies = moviesFromServer.filter(
+      (movie) => movie.title.toLowerCase()
+        .includes(searchQuery.trim().toLowerCase())
+        || movie.description.toLowerCase()
+          .includes(searchQuery.trim().toLowerCase()),
+    );
+
+    setVisibleMovies(filteredMovies);
+  };
+
+  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { value } = event.target;
+
+    setQuery(value);
+    filterMovies(value);
+  };
+
+  const memoizedVisibleMovies = useMemo(() => visibleMovies, [visibleMovies]);
```

Comment:
it's unnecessary to create one more movies list with such `useMemo`, but ypu can use `useCallback` to create a function, which depends on moviesFromServer 😉